packages/cli/src/commands/build.ts

```diff
@@ -148,6 +150,13 @@ export async function build({
 
   const resolver = overrideResolver || (await createDefaultReadRegistry(cliSettings));
 
+  const uri = await resolver.getUrl(fullPackageRef, chainId);
```

Comment:
2 possibly major issues here:
1. we only want to check that the package does not exist on the `OnChainRegistry`. It is not a problem if the package exists/is resolvable on the local registry (in fact, that is encouraged!)
2. we should not call `await resolver.getUrl` if `ignorePkgrefCheck` is set to true, since it is a completely wasted call